app/src/routes/api/v2/area.router.js

```diff
@@ -364,22 +400,31 @@ class AreaRouterV2 {
         ctx.statusCode = 204;
     }
 
-    static async getAlertsOfArea(ctx) {
-        logger.info(`Obtaining alerts of area with id ${ctx.params.id}`);
-        ctx.assert(ctx.query.precissionPoints, 400, 'precissionPoints is required');
-        ctx.assert(ctx.query.precissionBBOX, 400, 'precissionBBOX is required');
-        const result = await AreaModel.findOne({ _id: ctx.params.id });
-        if (!result) {
-            ctx.throw(404, 'Area not found');
+    static async updateByGeostore(ctx) {
+        if (ctx.state.loggedUser.role !== 'ADMIN') {
+            ctx.throw(401, 'Not authorized');
             return;
         }
-        let generateImages = true;
-        if (ctx.query.nogenerate) {
-            generateImages = false;
-        }
+        const geostores = ctx.request.body.geostores || [];
+        const updateParams = ctx.request.body.update_params || {};
+        logger.info('Updating geostores: ', geostores);
+        logger.info('Updating with params: ', updateParams);
+
+        // validate update json
+        updateParams.updatedDate = Date.now;
+
+        const response = await AreaModel.updateMany(
+            { geostore: { $in: geostores } },
+            { $set: updateParams }
+        );
 
-        const response = await AlertsService.groupAlerts(result, ctx.query.precissionPoints, ctx.query.precissionBBOX, generateImages);
-        ctx.body = response;
+        logger.info(`Updated ${response.nModified} out of ${response.n}.`);
+        if (response.ok && response.ok === 1) {
+            const areas = await AreaModel.find({ geostore: { $in: geostores } });
+            ctx.body = AreaSerializerV2.serialize(areas);
+        } else {
+            ctx.throw(404, 'Update failed.');
```

Comment:
"update failed" why? how are you informing the user about how to proceed to fix this? and should this really be a 404?

Reply:
I think this `else` clause might actually have never been used because if the update failed, an exception was thrown.

Replaced the `if-else` with a `try-catch` statement, and also a 400 Bad Request with an appropriate error message is sent in case of a failing update.